pkg/types/types.go

```diff
@@ -993,12 +995,12 @@ func (v Vsphere) GetSecret() map[string][]byte {
 }
 
 type Connection struct {
-	URL      string `yaml:"url" json:"url"`
-	User     string `yaml:"user,omitempty" json:"user,omitempty"`
-	Password string `yaml:"password,omitempty" json:"password,omitempty"`
-	Port     string `yaml:"port,omitempty" json:"port,omitempty"`
-	Scheme   string `yaml:"scheme,omitempty" json:"scheme,omitempty"`
-	Verify   string `yaml:"verify,omitempty" json:"verify,omitempty"`
+	URL      string         `yaml:"url" json:"url"`
+	User     string         `yaml:"user,omitempty" json:"user,omitempty"`
+	Password kommons.EnvVar `yaml:"password,omitempty" json:"password,omitempty"`
+	Port     string         `yaml:"port,omitempty" json:"port,omitempty"`
+	Scheme   string         `yaml:"scheme,omitempty" json:"scheme,omitempty"`
+	Verify   string         `yaml:"verify,omitempty" json:"verify,omitempty"`
```

Comment:
Probably makes sense to convert all of these to EnvVar

Reply:
I'm not sure - are the url/port/scheme likely to be in a configmap/secret?  In the case of elasticsearch the user is a key and not a value, so it can't be read from config either.  That might not be the case if we reuse connection elsewhere, but currently it's only used for *beat objects.